Create systemd setup for backend

decentralizedb opened the pull request:
If anyone wants to create a system service to run teos as a system service to start on the background, the instructions to create it are described in a new section in the install.md file.

sr-gi reviewed (commented):
Hey @decentralizedb, thanks for taking care of this.

I think this should be part of INSTALL.md instead of its own file.

It'll also be nice for this PR to have a minimal description.

meryacine commented:
Is `bitcoind.service` installed by default with bitcoin core?
If not I think you should publish it too (or put a hyperlink to it).

sr-gi commented on INSTALL.md:
```diff
@@ -14,4 +14,75 @@ You can run tests with:
+
+#### One you create it and copy and paste, change the information with your proper installation path, user, and comments:
+
+#### ------------- don't copy this line, copy under it --------------------------
```
I don't think this wrapping is really necessary. Given the file content is already formatted using the GH code formatter (```) it is pretty clear what needs to go in the file.

decentralizedb replied:
Ok, perfect, I removed wrapping and incorporated the parameters for logging in the system services file, please review it.

sr-gi commented on INSTALL.md:
````diff
@@ -14,4 +14,75 @@ You can run tests with:
+WantedBy=multi-user.target
+
+```
+#### ------------- don't copy this line, copy above it --------------------------
````
Remove

decentralizedb replied:
Done

sr-gi commented on INSTALL.md:
```diff
@@ -14,4 +14,75 @@ You can run tests with:
+Please refer to the cargo documentation for more detailed instructions.
+
+
+### If you want to create a system service to run your application follow the instructions below:
```
I'd move this inside the `Systemd setup for backend` section.

Also, If you want -> Should you want to / If you would like to

decentralizedb replied:
Done

sr-gi commented on INSTALL.md:
```diff
@@ -14,4 +14,75 @@ You can run tests with:
+# Systemd setup for backend
+
+
+#### Create a simple file copying the contents below to the file, and use the command:
```
Using you're preferred text editor, create a file in the `systemd` folder. For instance:

`sudo nano /etc/systemd/system/teosd.service`

decentralizedb replied:
All formatting and wording comments have been modified.

sr-gi commented on INSTALL.md:
```diff
@@ -14,4 +14,75 @@ You can run tests with:
+
+#### Create a simple file copying the contents below to the file, and use the command:
+`sudo nano /etc/systemd/system/teosd.service`
+#### This will create a file named teos.service in the /etc/systemd/system/ path
+
+#### One you create it and copy and paste, change the information with your proper installation path, user, and comments:
+
```
Copy the following information replacing the relevant information with yours (`<USER>` and `<GROUP>`):

decentralizedb replied:
Done

sr-gi commented on INSTALL.md:
````diff
@@ -14,4 +14,75 @@ You can run tests with:
+```
+#### ------------- don't copy this line, copy above it --------------------------
+
+#### The next step is to enable the service with the following command:
````
[...] is enabling the service. You can do so by running:

decentralizedb replied:
Done

sr-gi commented on INSTALL.md:
```diff
@@ -14,4 +14,75 @@ You can run tests with:
+#### The next step is to enable the service with the following command:
+`sudo systemctl enable teosd.service`
+
+#### And start the service with the command:
```
Finally, you can start the service by running:

decentralizedb replied:
Done

sr-gi commented on INSTALL.md:
```diff
@@ -14,4 +14,75 @@ You can run tests with:
+
+#### And start the service with the command:
+`sudo systemctl start teosd.service`
+
+#### If you need to stop the service, use the command:
+`sudo systemctl stop teosd.service`
```
I don't think this is really necessary

decentralizedb replied:
Removed

sr-gi reviewed (changes requested):
Some nits.

As a general comment, I don't think the text needs to be bolded for all paragraphs (####).

sr-gi commented:
> 

Agreed, if this depends on another service it would be nice to have a link to it from a well know guide (such as raspibolt)

sr-gi commented:
> Is `bitcoind.service` installed by default with bitcoin core? If not I think you should publish it too (or put a hyperlink to it).

Agreed, if this depends on another service it would be nice to have a link to it from a well know guide (such as raspibolt)

decentralizedb commented:
> Is `bitcoind.service` installed by default with bitcoin core? If not I think you should publish it too (or put a hyperlink to it).

I added the link to a guide that is quite detailed and updated instructions. 
This guide has just been released and is quite good in and installation oriented on how to set up on most hardware platforms because it only uses standard Debian Linux commands, as the guide indicates.

decentralizedb commented:
I attended all indications, I hope this time it will be ok, but if not, please let me know. I'm happy to help and learn!  xD

sr-gi commented:
Tagging @openoms here since he's way more knowledgeable than me

sr-gi reviewed (changes requested):
I think this makes sense, but I'd structure it differently. I think we can follow the same structure followed by Bitcoin Core for their init files (check https://github.com/bitcoin/bitcoin/blob/master/contrib/init and https://github.com/bitcoin/bitcoin/blob/master/contrib/init/bitcoind.service), that is:

- [ ]  Create a new folder in the project root called `contrib`
    - [ ] Create a subfolder called `init`
    - [ ] Create a `README.md` with most of the content added to this file in the PR
    - [ ] Create a `teosd.service` file including the relevant service information
- [ ] Link the `contrib/init/README.md` file to `INSTALL.md` under the `Systemd setup for backend`

This may like:

```
[...]
Please refer to the cargo documentation for more detailed instructions.

# Systemd setup for backend
Refer to init(LINK) for a detailed explanation of how to set up your systemd service or something on these lines.
```
